internal/services/access_flow.go

```diff
@@ -0,0 +1,476 @@
+	if len(diagnostics) > 0 {
+		return nil, diagnostics
+	}
+
+	var dataApprovers []apono.ApproverV1
+	for _, approver := range existingApprovers {
+		approverIds, diagnostics := getIdentitiesIdsByNameAndType(ctx, approver.Name.ValueString(), approver.Type.ValueString(), availableIdentities.Data, aponoClient)
+		if len(diagnostics) > 0 {
+			return nil, diagnostics
+		}
+
+		for _, approverId := range approverIds {
+			dataApprovers = append(dataApprovers, apono.ApproverV1{
+				Id:   approverId,
+				Type: approver.Type.ValueString(),
+			})
+		}
+	}
+	data.Approvers = dataApprovers
+
+	availableIntegrations, _, err := aponoClient.IntegrationsApi.ListIntegrationsV2(ctx).Execute()
+	if err != nil {
+		return nil, utils.GetDiagnosticsForApiError(err, "list", "integrations", "")
+	}
+
+	var dataIntegrationTargets []apono.AccessTargetIntegrationV1
+	for _, integrationTarget := range accessFlow.IntegrationTargets {
+		dataIntegrationTarget, diagnostics := convertIntegrationTargetToApiModel(integrationTarget, availableIntegrations.Data)
+		if len(diagnostics) > 0 {
+			return nil, diagnostics
+		}
+
+		dataIntegrationTargets = append(dataIntegrationTargets, *dataIntegrationTarget)
+	}
+	data.IntegrationTargets = dataIntegrationTargets
+
+	if accessFlow.Settings != nil {
+		settings := apono.AccessFlowV1Settings{
+			RequireJustificationOnRequestAgain: *apono.NewNullableBool(accessFlow.Settings.RequireJustificationOnRequestAgain.ValueBoolPointer()),
+			RequireAllApprovers:                *apono.NewNullableBool(accessFlow.Settings.RequireAllApprovers.ValueBoolPointer()),
+			ApproverCannotApproveHimself:       *apono.NewNullableBool(accessFlow.Settings.ApproverCannotApproveHimself.ValueBoolPointer()),
+		}
+
+		data.Settings = *apono.NewNullableAccessFlowV1Settings(&settings)
+	}
+
+	return &data, nil
+}
+
+func ConvertToAccessFlowUpdateApiModel(ctx context.Context, aponoClient *apono.APIClient, accessFlow *models.AccessFlowModel) (*apono.UpdateAccessFlowV1, diag.Diagnostics) {
+	updateAccessFlowRequest, diagnostics := ConvertToAccessFlowUpsertApiModel(ctx, aponoClient, accessFlow)
+	if len(diagnostics) > 0 {
+		return nil, diagnostics
+	}
+
+	trigger := apono.UpdateAccessFlowV1Trigger{
+		Type:      updateAccessFlowRequest.Trigger.Type,
+		Timeframe: updateAccessFlowRequest.Trigger.Timeframe,
+	}
+
+	var approvers []apono.ApproverV1
+	if updateAccessFlowRequest.Approvers == nil {
+		approvers = make([]apono.ApproverV1, 0)
+	} else {
+		approvers = updateAccessFlowRequest.Approvers
+	}
+
+	data := apono.UpdateAccessFlowV1{
+		Name:               *apono.NewNullableString(&updateAccessFlowRequest.Name),
+		Active:             *apono.NewNullableBool(&updateAccessFlowRequest.Active),
+		RevokeAfterInSec:   *apono.NewNullableInt32(&updateAccessFlowRequest.RevokeAfterInSec),
+		Trigger:            *apono.NewNullableUpdateAccessFlowV1Trigger(&trigger),
+		Grantees:           updateAccessFlowRequest.Grantees,
+		Approvers:          approvers,
```

Comment:
why this difference in their handling?

Reply:
approver list is optional field so it can be null, and if it null i replace it with empty list